internal/cloudfoundry/initialize.go

```diff
@@ -20,11 +20,12 @@ type InitializePhase interface {
 }
 
 type Initialize struct {
-	cli Executable
+	cli   Executable
+	stack string
 }
 
-func NewInitialize(cli Executable) Initialize {
-	return Initialize{cli: cli}
+func NewInitialize(cli Executable, stack string) Initialize {
+	return Initialize{cli: cli, stack: stack}
 }
 
 func (i Initialize) Run(buildpacks []Buildpack) error {
```

Comment:
It seems a little odd to me that the proposed change initializes the Cloudfoundry platform with a `cli` and a `stack` - they're very different objects, with different levels of abstraction. When I see `stack`, I wonder why I don't also see `buildpacks`.

I wonder if it makes more sense to change the `Run` function signature to include the stack here, rather than at initialization time?

What do you think? Am I missing something?

Reply:
I agree with you that they are totally different objects, but semantically the initialization process of the platform Buildpacks requires this stack, so it could be justified.

On the other hand, moving this parameter to Run(), we would have to modify the tests in the Buildpacks that have Switchblade to support this parameter, which would make this change require additional changes to all the Buildpacks.

Reply:
I agree with you that they are totally different objects, but semantically the initialization process of the platform Buildpacks requires this stack, so it could be justified.

On the other hand, moving this parameter to Run(), we would have to modify the tests in the Buildpacks that have Switchblade to support this parameter, which would make this change require additional changes to all the Buildpacks.

Reply:
Ok, I'm convinced that this change is fine. I see that the interface of `platform.Initialize` - which is what is exposed to buildpacks' tests - only has buildpacks. I also see that the `Docker` `case` also does something similar to this PR. So we're already encapsulating the stack in the setup.

If I was creating this from scratch, I might not write this, but I think as an incremental change it's fine.